### PR TITLE
clean AMP logic

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

seems that AMP logic and warning was fishy...

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/models/test_amp.py [hunk 1]
```diff
@@ -155,21 +155,12 @@ def test_amp_gpu_ddp_slurm_managed(tmpdir):
     assert generated == 'abc23'
 
 
+@mock.patch('apex.amp', return_value=None)
```
Comment (Borda) [inline tests/models/test_amp.py:158]:
```suggestion
```

File: pytorch_lightning/accelerators/accelerator_connector.py [hunk 2]
```diff
@@ -298,43 +298,40 @@ def select_precision_plugin(self) -> PrecisionPlugin:
...
+                        raise MisconfigurationException(msg)
                 else:
                     log.info("Using native 16bit precision.")
                     if isinstance(self.training_type_plugin, (DDPShardedPlugin, DDPSpawnShardedPlugin)):
                         return ShardedNativeMixedPrecisionPlugin()
-                    self.amp_type = AMPType.NATIVE
                     return NativeMixedPrecisionPlugin()
 
-            if self.amp_type == "apex":
+            if self.amp_type == AMPType.APEX:
```
Comment (tchaton) [inline pytorch_lightning/accelerators/accelerator_connector.py:322]: Is Apex available for cpu ?

**Review by tchaton — APPROVED** [review]
LGTM ! Nice cleaning !

File: pytorch_lightning/accelerators/accelerator_connector.py (hunk 2 above)
Comment (Borda) [reply pytorch_lightning/accelerators/accelerator_connector.py:322]: that was my question, according to our actual code and @SeanNaren yes

Comment (Borda) [reply pytorch_lightning/accelerators/accelerator_connector.py:322]: but yes this was very hard to follow what it does so I kicke-up cleaning :D

File: pytorch_lightning/accelerators/accelerator_connector.py [hunk 3]
```diff
@@ -298,43 +298,40 @@ def select_precision_plugin(self) -> PrecisionPlugin:
...
                 if not _APEX_AVAILABLE:
-                    rank_zero_warn(
+                    raise MisconfigurationException(
                         "You have asked for Apex AMP but you have not installed it yet."
                         " Install apex first using this guide: https://github.com/NVIDIA/apex#linux"
                     )
                 else:
```
Comment (carmocca) [inline pytorch_lightning/accelerators/accelerator_connector.py:328]: Can you also remove this else? It should be unnecessary

File: pytorch_lightning/accelerators/accelerator_connector.py [hunk 4]
```diff
@@ -298,43 +298,40 @@ def select_precision_plugin(self) -> PrecisionPlugin:
...
                         raise MisconfigurationException(
-                            "Sharded Plugin is not supported with Apex AMP, "
-                            "please using native AMP for 16-bit precision."
+                            "Sharded Plugin is not supported with Apex AMP,"
+                            " please using native AMP for 16-bit precision."
                         )
                     log.info("Using APEX 16bit precision.")
-                    self.amp_type = AMPType.APEX
                     return ApexMixedPrecisionPlugin(self.amp_level)
         else:
```
Comment (carmocca) [inline pytorch_lightning/accelerators/accelerator_connector.py:336]: Can you also remove this else? It should be unnecessary

File: pytorch_lightning/accelerators/accelerator_connector.py (hunk 3 above)
Comment (Borda) [reply pytorch_lightning/accelerators/accelerator_connector.py:328]: what do you mean?

**Review by carmocca — APPROVED** [review]

File: pytorch_lightning/accelerators/accelerator_connector.py (hunk 2 above)
Comment (SeanNaren) [reply pytorch_lightning/accelerators/accelerator_connector.py:322]: It's not straight forward I think, I tried doing `pip install .` locally and it seems it looks for CUDA_HOME to compile things with... I'd say no need for CPU support

**Review by SeanNaren — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]
